sdv/metadata/single_table.py

```diff
@@ -250,6 +290,19 @@ def to_dict(self):
 
         return deepcopy(metadata)
 
+    def _detect_pii_column(self, column_name):
+        """Detect PII columns.
+
+        Args:
+            column_name (str):
+                The column name to be analyzed.
+        """
+        cleaned_name = re.sub(r'[^a-zA-Z0-9]', '', column_name).lower()
+        return next((
+            sdtype for reference, sdtype in self._REFERENCE_TO_SDTYPE.items()
+            if cleaned_name == reference or reference in cleaned_name
```

Comment:
if reference == cleaned_name is true, so is reference in cleaned_name so I think you only need the second check